File: Proyecto en C#/Entrega Diseno/Sector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data.OleDb;

namespace Entrega_Diseno
{
    class Sector
    {
        private string _Nombre;
        public string Nombre
        {
            get
            {
                return _Nombre;
            }
            set
            {
                _Nombre = value;
            }
        }

        private int _Precio;
        public int Precio
        {
            get { return _Precio; }
            set { _Precio = value; }
        }

        private RangoFilas _RangoFilas;
        public RangoFilas RangoFilas
        {
            get { return _RangoFilas; }
            set { _RangoFilas = value; }
        }

        public Sector(string Nombre)
        {

            _Nombre = Nombre;
            if (_Nombre == "Sector:A")
            {
                _Precio = 320;
            }
            else
            {
                _Precio = 100;
            }
        }

        public float valorEntradaBase()
        {
            return (Precio + _RangoFilas.Precio);
        }
    }
}

File: recital-dominio/src/main/scala/recital/Sector.scala
package recital

import uqbar.arena.persistence.annotations.PersistentField
import uqbar.arena.persistence.annotations.Relation
import uqbar.arena.persistence.annotations.PersistentClass
import compatibility.DomainEntity

@PersistentClass
class Sector extends DomainEntity {
  
  def this(nombre: String) = {
    this()
    this.nombre = nombre
  }
  
  var nombre: String =_
  var filas: Set[RangoFilas] = Set()
  
  @PersistentField
  def getNombre = nombre
  def setNombre(n:String) = nombre = n
  
  @Relation
  def getFilas : java.util.List[RangoFilas] = setToList(filas)
  def setFilas(f:java.util.List[RangoFilas]) = filas = listToSet(f)
	
  
  def valorEntradaBase(numero: Int): Double ={
    val fila = buscarFila(numero)
    	
    fila.precio
  }
  
  def verificarUbicacion(fila:Int, asiento:Int) = {
    val filaEncontrada = buscarFila(fila)
    
    if( !(filaEncontrada.existeAsiento(asiento)) ){
      throw new AsientoIncorrectoException
    }
  }
  
  def buscarFila(numero: Int): RangoFilas = {
    var fila = filas.find{fila: RangoFilas => fila.tengoFila(numero)}
    
    if (fila.isEmpty) 
    	throw new FilaIncorrectaException      
    	
    fila.get
  }
  
 
}